clients/pkg/logentry/stages/metrics_test.go

```diff
@@ -127,6 +127,13 @@ func TestMetricsPipeline(t *testing.T) {
 		strings.NewReader(expectedMetrics)); err != nil {
 		t.Fatalf("mismatch metrics: %v", err)
 	}
+
+	pl.Cleanup()
+
+	if err := testutil.GatherAndCompare(registry,
+		strings.NewReader("")); err != nil {
+		t.Fatalf("mismatch metrics: %v", err)
+	}
```

Comment:
shouldn't we specifically check for the absence of the metric?

Reply:
After `pl.Cleanup()`, no metrics should be visible. I don't see an advantage in checking individual metrics when there shouldn't be any of them.

Reply:
what if there's originally two metrics stages, and after we reload there should be only one? we should still check for it's existence and correct value, I think right now we're deleting metrics for stages that will still exist after the reload which would not be correct

Reply:
I'd argue that deleting all the metrics is the right thing to do. The fact that a metric exists doesn't mean that it's used in the same way in the new config. It's possible that the user changed the meaning of each metric, but retained the metric names. In that case it would make sense to start with fresh metrics.

In the future we could make the code smart and reload only the stages which changed, but I think that's a separate issue. To do that, we could avoid calling the `Cleanup` method for stages which are completely unchanged as a whole.

Reply:
we should document this clearly in the metrics config section https://github.com/grafana/loki/blob/a00f1f1b0b8f536f2cdac2f8857eb40c716aa696/docs/sources/send-data/promtail/configuration.md?plain=1#L678 and maybe as a note here as well